java/fury-core/src/main/java/io/fury/collection/ImmutableSet.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 The Fury Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fury.collection;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class ImmutableSet {
```

Comment:
Could we use java.util.collection at the invoke point instead? Add a new public class for guava only may make new users and developers confused.

Reply:
You don't want to introduce `ImmutableSet` & `ImmutableList`? Instead use `java.util.collection` directly?

Reply:
Yep, ImmutableSet and ImmutableList is used widely in test code, but barely used in source code. We can use jdk collection instead. In the end, there will no guava dependency in fury-core module.